src/rdbms/azext_rdbms/custom.py

```diff
@@ -31,143 +86,14 @@ def _server_restore(cmd, client, resource_group_name, server_name, parameters, n
             raise ValueError('The provided source-server {} is invalid.'.format(source_server))
 
     parameters.properties.source_server_id = source_server
+    parameters.location = location
 
-    # Here is a workaround that we don't support cross-region restore currently,
-    # so the location must be set as the same as source server (not the resource group)
-    id_parts = parse_resource_id(source_server)
+    source_server_id_parts = parse_resource_id(source_server)
     try:
-        source_server_object = client.get(id_parts['resource_group'], id_parts['name'])
-        parameters.location = source_server_object.location
+        source_server_object = client.get(source_server_id_parts['resource_group'], source_server_id_parts['name'])
+        if parameters.sku.name is None:
+            parameters.sku.name = source_server_object.sku.name
     except Exception as e:
         raise ValueError('Unable to get source server: {}.'.format(str(e)))
 
     return client.create(resource_group_name, server_name, parameters, raw=no_wait)
```

Comment:
`raw=no_wait` no longer works for newer SDKs that are generated with Autorest 3.0. The polling will still happen which is not the intent here.

Instead
```
from azure.cli.core.util import sdk_no_wait
return sdk_no_wait(no_wait , client.create, resource_group_name, server_name, parameters)
```

You can see "(4) Supporting --no-wait" in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#register-commands